target/linux/mediatek/patches-5.4/0402-net-bridge-fix-client-roaming-from-DSA-user-port.patch

```diff
@@ -0,0 +1,49 @@
+From 269e743843bd94799bb61c09538cc6c7fdeab616 Mon Sep 17 00:00:00 2001
+From: DENG Qingfang <dqfext@gmail.com>
+Date: Sun, 19 Apr 2020 23:38:46 +0800
+Subject: [PATCH] net: bridge: fix client roaming from DSA user port
+
+When a client roams from a DSA user port to a soft-bridged port (such as
+WiFi interface), the left-over MAC entry in the switch HW is not deleted,
+causing inconsistency between Linux fdb and the switch MAC table. As a
+result, the client cannot talk to other hosts which are on that DSA user
+port until the MAC entry expires.
+
+Solve this by notifying switchdev fdb to delete the leftover entry when
+an entry is updated. Remove the added_by_user check
+
+Signed-off-by: DENG Qingfang <dqfext@gmail.com>
+---
+ net/bridge/br_fdb.c | 2 ++
+ net/dsa/slave.c     | 2 --
+ 2 files changed, 2 insertions(+), 2 deletions(-)
```

Comment:
This is a patch for generic code. If you believe the concept in this patch is correct, you don't need to duplicate the patch in two targets.

Reply:
(And if you added the patch to two different targets, it should be one commit per target then ...)

Reply:
I already stated the reason in commit message.

Reply:
@dengqf6 I'm aware of your commit message.
If the patch is correct, it should be the marvell driver's fault that the problem isn't solved there. You don't need to put that patch in two targets just because of this.

Reply:
What if it caused problems on other targets?

Reply:
Just FYI on my TP-Link RE650 (mt7621) this patch works 100%.

I removed the "auto learn CPU port" patch and patched using this and am having no issues roaming.